### PR TITLE
Enemies branch

### PR DESCRIPTION
**Pull request by rojtekin**
Enemies user-story complete

### CODE REVIEW

File: core/src/Controller/Controller.java [hunk 1]
```diff
@@ -8,9 +8,9 @@ public class Controller {
     private IModel model;
     private IControllable player;
 
-    public Controller(IModel model, IControllable player) {
+    public Controller(Model model) {
         this.model = model;
-        this.player = player;
+        this.player = model.getPlayer();
```
Comment (pkasinski) [inline core/src/Controller/Controller.java:13]: Does Controller need a player?

File: core/src/Model/Enemies/Mouse.java [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+import com.badlogic.gdx.Gdx;
 
 public class Mouse extends Enemy {
 
-    public Mouse(int x, int y, int height, int width, int speed, float health, float damage) {
+    public Mouse(int x, int y, int height, int width, int speed, double health, double damage) {
         super(x, y, height, width, speed, health, damage);
-        setImagePath("Enemies/mouse-idle-down.png");
+        setEntityName("Mouse");
+        addSound(Gdx.audio.newSound(Gdx.files.internal("Enemies/Mouse/mouse-squeek.mp3")));
```
Comment (pkasinski) [inline core/src/Model/Enemies/Mouse.java]: Should sound be added here or in View?

**Review by Suedie — APPROVED** [review]
LGTM

File: core/src/Controller/Controller.java (hunk 1 above)
Comment (pkasinski) [inline core/src/Controller/Controller.java:13]: Does Controller need a player?

File: core/src/Model/Enemies/Mouse.java [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 import Model.Enemy;
+import com.badlogic.gdx.Gdx;
 
 public class Mouse extends Enemy {
 
-    public Mouse(int x, int y, int height, int width, int speed, float health, float damage) {
+    public Mouse(int x, int y, int height, int width, int speed, double health, double damage) {
         super(x, y, height, width, speed, health, damage);
-        setImagePath("Enemies/mouse-idle-down.png");
+        setEntityName("Mouse");
```
Comment (pkasinski) [inline core/src/Model/Enemies/Mouse.java:9]: Should this not be a part of the constructor?

Comment (pkasinski) [reply core/src/Model/Enemies/Mouse.java:9]: Is EntityName needed?

File: core/src/Model/Enemy.java [hunk 4]
```diff
@@ -1,17 +1,11 @@
 package Model;
 
 public abstract class Enemy extends Entity{
-    private float damage;
-    private String imagePath = ""; // hur gör man att de måste finnas en?
+    private double damage;
+    private String enemyName = ""; // Should be the same as its corresponding prefix under assets.
```
Comment (pkasinski) [inline core/src/Model/Enemy.java:5]: Is both enemyName and entityName needed?

File: core/src/Model/Entity.java [hunk 5]
```diff
@@ -1,35 +1,47 @@
...
 private int y;
 private int height;
 private int width;
 private int speed;
-private float health;
+private double health;
+private boolean inMotion = false;
+private String entityName;
+private int angle;
+private ArrayList<com.badlogic.gdx.audio.Sound> sounds = new ArrayList<>();
```
Comment (pkasinski) [inline core/src/Model/Entity.java]: Should sounds be a part of the model?

File: core/src/View/Sound.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import com.badlogic.gdx.Gdx;
+
+import java.util.ArrayList;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+
+public class Sound{
```
Comment (pkasinski) [inline core/src/View/Sound.java:15]: This name can be confusing. LibGDX has a class named Sound.

File: core/src/View/Sound.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    public com.badlogic.gdx.audio.Sound getSound() { //TODO generalise for everything
+
+        com.badlogic.gdx.audio.Sound sound = Gdx.audio.newSound(Gdx.files.internal("Enemies/Mouse/mouse-squeek.mp3"));
+        return sound;
+    }
+
+    public void playSounds(final Model model) {
```
Comment (pkasinski) [inline core/src/View/Sound.java:22]: Documentation for this method could be added.

**Review by pkasinski — APPROVED** [review]
Looks good. I left a few comments with questions and suggestions.

File: core/src/Model/Enemy.java (hunk 4 above)
Comment (rojtekin) [reply core/src/Model/Enemy.java:5]: yes entityName = name of species 
enemyName = individual name of the enemy

File: core/src/View/Sound.java (hunk 6 above)
Comment (rojtekin) [reply core/src/View/Sound.java:15]: in order to access that class you have to use "com.badlogic.gdx.audio.Sound" which is pain just using "Sound" is much nicer

File: core/src/View/Sound.java (hunk 7 above)
Comment (rojtekin) [reply core/src/View/Sound.java:22]: Yes sir

**Pull request merged**